src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double length;
+    private double width;
+
+    public Rectangle(String color, double length, double width) {
+        super(color);
+        this.length = length;
+        this.width = width;
+    }
+
+    public String toString() {
+        return "Figure: rectangle, area: " + calculateArea() + " sq. units, "
+                + "length: " + length + " units, "
+                + "width: " + width + " units, "
+                + "color: " + getColor();

```

Comment:
Use the `name()` method for getting the String representation of enum constants. If `getColor()` returns an enum value, make sure to call `getColor().name()` instead of `getColor()` directly in the `toString()` method.